packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/file.ts

```diff
@@ -0,0 +1,372 @@
+import type { AntiRaidResult } from '../../anti-raid-nuke.js';
+
+export interface AntiRaidFileArgs {
+	file: Attachment;
+	reason?: string | undefined;
+	days?: number | undefined;
+	hide?: boolean | undefined;
+}
+
+async function parseFile(file: Attachment): Promise<Set<string>> {
+	const content = await fetch(file.url).then((res) => res.text());
+
+	const ids: string[] | null = content.match(/\d{17,20}/g);
+
+	if (!ids?.length) {
+		return new Set();
+	}
+
+	return new Set(ids);
+}
+
+export async function file(
+	interaction: CommandInteraction<'cached'>,
+	data: AntiRaidFileArgs,
+	logChannel: TextChannel,
+	ignoreRolesId: string[],
+	locale: string,
+	redis: Redis,
+): Promise<void> {
+	const start = performance.now();
+
+	const reply = await interaction.deferReply({ ephemeral: data.hide ?? true });
+
+	const file = data.file;
+	const ids = await parseFile(file);
+
+	if (!ids.size) {
+		throw new Error(i18next.t('command.mod.anti_raid_nuke.errors.no_ids', { lng: locale }));
+	}
+
+	const { reason, days } = {
+		reason: data.reason ?? null,
+		days: data.days ?? 1,
+	};
+
+	const fetchedMembers = await interaction.guild.members.fetch();
+	const members = new Collection<string, GuildMember>();
+	const fails = new Set<string>();
+
+	for (const id of ids) {
+		const member = fetchedMembers.get(id);
+		if (member) {
+			members.set(id, member);
+		} else {
+			fails.add(id);
+		}
+	}
+
+	if (!members.size) {
+		throw new Error(i18next.t('command.mod.anti_raid_nuke.errors.no_hits_file', { lng: locale }));
+	}
+
+	const parameterStrings = [
+		i18next.t('command.mod.anti_raid_nuke.parameters.heading', { lng: locale }),
+		i18next.t('command.mod.anti_raid_nuke.parameters.current_time', {
+			lng: locale,
+			now: Formatters.time(dayjs().unix(), Formatters.TimestampStyles.ShortDateTime),
+		}),
+		i18next.t('command.mod.anti_raid_nuke.parameters.file', {
+			lng: locale,
+			file: Formatters.hyperlink('File uploaded', file.url),
+		}),
+	];
+
+	if (fails.size) {
+		parameterStrings.push(
+			i18next.t('command.mod.anti_raid_nuke.parameters.users', {
+				lng: locale,
+				users: Formatters.inlineCode(fails.size.toString()),
+			}),
+		);
+	}
+
+	const banKey = nanoid();
+	const cancelKey = nanoid();
+	const dryRunKey = nanoid();
+
+	const banButton = createButton({
+		customId: banKey,
+		label: i18next.t('command.mod.anti_raid_nuke.buttons.execute', { lng: locale }),
+		style: ButtonStyle.Danger,
+	});
+	const cancelButton = createButton({
+		customId: cancelKey,
+		label: i18next.t('command.mod.anti_raid_nuke.buttons.cancel', { lng: locale }),
+		style: ButtonStyle.Secondary,
+	});
+	const dryRunButton = createButton({
+		customId: dryRunKey,
+		label: i18next.t('command.mod.anti_raid_nuke.buttons.dry_run', { lng: locale }),
+		style: ButtonStyle.Primary,
+	});
+
+	const potentialHits = Buffer.from(members.map((member) => generateTargetInformation(member)).join('\r\n'));
```

Comment:
```suggestion
	const potentialHits = Buffer.from(members.map((member) => generateTargetInformation(member)).join('\n'));
```

Shouldn't use a carriage return here.

Reply:
Implemented carriage return on all aplicable `.join()`s